### ISSUE TITLE
ggplot2 Extension

### ISSUE THREAD

**Issue by vikjam** [paulgp/binscattr#6]
Not sure if it'd be helpful, but have you looked at implementing this as a [ggplot2 extension](https://cran.r-project.org/web/packages/ggplot2/vignettes/extending-ggplot2.html)?

**Comment by paulgp** [comment]
Yes, this could work -- we're effectively using stat_summary and
geom_smooth anyway. The only complicated part is doing the residualization
in the background.

**Comment by vikjam** [comment]
Cool, I'll take a closer look at this and put a pull request together (if that'd be helpful).

**Comment by paulgp** [comment]
That would be great!
